tmt/utils/__init__.py

```diff
@@ -230,13 +230,16 @@ def format(cls, exit_code: int) -> Optional[str]:
 T = TypeVar('T')
 
 
-def effective_workdir_root() -> Path:
+def effective_workdir_root(workdir_root_option: Optional[str] = None) -> Path:
```

Comment:
I'd go with `Optional[Path]` already: as soon as we get the value of `--workdir-root` option in CLI commands, we should convert it to either a `None` or `Path` instance.